lib/diagnostic.js

```diff
@@ -6,23 +6,38 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (paragraph) {
+  for (let i = 0; i < words.length; i++) {
+  let word = words[i].split(/\s+/).replace(/\W+/, '');
+  normalizedWords[i] = word.toUpperCase();
+  }
+  return paragraph;
 };
```

Comment:
You are returning `paragraph`, the variable for your passed-in argument, but has it been modified in any way? Please take a look at this one possible solution:
```
const normalizeWords = function (text) {
  let words = text.split(/\s+/);
  for (let i = 0; i < words.length; i++) {
    let word = words[i];
    words[i] = word.replace(/\W+/, '').toUpperCase();
  }
  return words;
};
```